lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1089,10 +1099,24 @@ def deploy_vm(self):
                 relospec.datastore = datastore
                 relospec.pool = resource_pool
 
+                if self.params['snapshot_src'] is not None and self.params['linked_clone']:
+                    relospec.diskMoveType = vim.vm.RelocateSpec.DiskMoveOptions.createNewChildDiskBacking
+
                 clonespec = vim.vm.CloneSpec(template=self.params['is_template'], location=relospec)
                 if self.customspec:
                     clonespec.customization = self.customspec
 
+                if self.params['snapshot_src'] is not None:
+                    snapshot = self.get_snapshots_by_name_recursively(snapshots=vm_obj.snapshot.rootSnapshotList,
+                                                                      snapname=self.params['snapshot_src'])
+                    if len(snapshot) != 1:
+                        self.module.fail_json(msg='virtual machine "{0}" does not contain snapshot named "{1}"'.format(
+                            self.params['template'], self.params['snapshot_src']))
+
+                    if len(snapshot) == 1:
```

Comment:
you can remove this condition as the previous condition already handle other cases